lib/dor/services/client/object.rb

```diff
@@ -111,6 +112,28 @@ def current_version
           raise_exception_based_on_response!(resp)
         end
 
+        # Determines if a new version can be opened for a DOR object.
+        # @raise [NotFoundResponse] when the response is a 404 (object not found)
+        # @raise [UnexpectedResponse] when the response is not successful.
+        # @return [Boolean] true if a new version can be opened
+        # rubocop:disable Metrics/MethodLength
+        def openeable?
```

Comment:
I think this should be in a class that is about the versions of an object, not the object itself.

Reply:
I think you should move `current_version`, `open_new_version`, `close_version` to that class too and delegate methods there.